### ISSUE TITLE
Usernames in death messages not parsed correctly when not followed by a space

### ISSUE THREAD

**Issue by unilock** [fooooooooooooooo/yep-paper#16]
The death message parsing logic doesn't account for when the username is followed by anything other than a space.

For example, the Twilight Forest mod adds the following death message for one of its bosses:
```
PLAYER's skin was ripped off by the Hydra
```
...which appears as the following in Discord:
```
PLAYER PLAYER's skin was ripped off by the Hydra
```

I'm not sure what the most elegant fix for this would be. It could just check for an apostrophe *or* a space, I suppose - I can't think of any other characters that might need to be checked for, off the top of my head.

**Comment by fooooooooooooooo** [comment]
in its current state, yep processes messages like this

`PLAYER was squished too much` > `was squished too much`
`PLAYER walked into a cactus whilst trying to escape PLAYER2` > `walked into a cactus whilst trying to escape PLAYER2`

and if it matches `'s` as well:

`PLAYER's skin was ripped off by the Hydra` > `skin was ripped off by the Hydra`

this works so far as you can just prepend the player name passed with the DeathMessage

in the case of a message formatted like this

`PLAYER was x by PLAYER doing y`

it would become

`was x by doing y`

then after prepending the player name on the consumer side

`PLAYER was x by doing y`

which probably wouldn't make much sense

this current format also doesn't allow for many options of what to do with the message as you only have the two strings `"PLAYER"` and `"was squished too much"`

so for a solution, yep could just not modify the message and send `"PLAYER"` and `"PLAYER was squished too much"`

this removes the chance for yep to mangle the message but doesn't solve the configurability issue

alternatively yep could return the parameterized message

for example using `death.attack.arrow.item` from mc 1.20:

`%1$s was shot by %2$s using %3$s`

if yep could extract the parameters from the message and send them along with `<player_1> was shot by <player_2> using <item_1>`, this would allow you to modify the parameterized text or just throw it away and write your own message format using the provided parameters

in velocitydiscord you could have a config option for how to format players and items:

```toml
death_message_player = "**{username}**"
death_message_item = "*{name}*"
death_message = "{death_message}"
```

to make player names bold and items italic for example

you could also have an option to throw away the original message:

```toml
use_original_death_message = false # default to true possibly
death_message_single = "{player} {items}"
death_message_multi = "{player_1} - {player_2} {items}"
```

would take a bit more work and testing to come up with an easy to use config for this case

the fact that there isnt a hard limit that i know of to the number of players or items makes this usage a bit more difficult

extracting the parameters from the message might not be easily done, especially across versions

so in summary:

* sending just `DEATH:PLAYER:PLAYER was squished too much` is the easiest option, but leaves little room for configuration
* sending something like `DEATH:PLAYER:<player_1> was shot by <player_2> using <item_1>|PLAYER|PLAYER2|Wicked Item` is potentially really difficult, but allows for a really customizable experience, which was my whole reason for creating velocitydiscord

**Comment by unilock** [comment]
> so for a solution, yep could just not modify the message and send `"PLAYER"` and `"PLAYER was squished too much"`

On that note, I recall in a heavily modded environment some death messages appearing differently between the in-game chat and Discord - I believe the culprit there was [Death Knell](https://github.com/Darkhax-Minecraft/Death-Knell).

Maybe a callback could be injected into `getDeathMessage()` or what have you, and both the method parameters (translation key + parameters (username, etc)) and the returned death message could be forwarded to Velocity via yep?

There's likely some way to get the translation key as a template, too - although it would be difficult to comprehensively account for this via a Velocity-side config file, since translation keys can have an arbitrary number of parameters of all different types.  
Maybe yep could send the (remapped) class names of the parameters, too? Sounds like a hassle... (i.e. "PlayerEntity" instead of "arg1", "LivingEntity" instead of "arg2", and "ItemStack" instead of "arg3" in the key `"<player> was slain by <entity> using <item>"`)

In my opinion, I think `DEATH:PLAYER:PLAYER was squished too much` would be the most efficient solution; I can't think of a case where the end user would want to completely discard the original death message when forwarding it to Discord. But that's just me.
